domain/keymanager/state/state.go

```diff
@@ -224,6 +225,55 @@ WHERE user_id = $userId.user_id
 		)
 	}
 
+	rval := make([]coressh.PublicKey, 0, len(publicKeys))
+	for _, pk := range publicKeys {
+		rval = append(rval, coressh.PublicKey{
+			Fingerprint: pk.Fingerprint,
+			Key:         pk.PublicKey,
+		})
+	}
+	return rval, nil
+}
+
+// GetPublicKeysDataForUser is responsible for returning all of the public keys
+// raw data for the user id in this model. If the user does not exist no error
+// is returned.
+func (s *State) GetPublicKeysDataForUser(
+	ctx context.Context,
+	id user.UUID,
+) ([]string, error) {
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	stmt, err := s.Prepare(`
+SELECT (public_key) AS (&publicKeyData.*)
+FROM user_public_ssh_key
+WHERE user_id = $userId.user_id
+`, userId{}, publicKeyData{})
+
+	if err != nil {
+		return nil, fmt.Errorf(
+			"preparing select statement for getting public keys data of user %q: %w",
```

Comment:
Our standard pattern is not to annotate `s.Prepare` errors. These would be emitted when unit tests are run and are never user facing. To do something different just in this one spot isn't something we'd want. If we want to change he approach, we'd need to discuss and do it everywhere.

Reply:
Happy to change but my reason was to figure out the problem and not play guess that broken sql statement.

Reply:
I find that just `errors.Trace()` is sufficient to track down the issue during development - and as you say, it's not something we would ever see in production and is relatively rare during development even. YMMV but consistency is how `s.Prepare()` is called across the various state packages is what I'd aim for given the above statements.